src/linux/cgroups2.hpp

```diff
@@ -275,6 +275,24 @@ Try<Nothing> set_max(const std::string& cgroup, const Option<Bytes>& limit);
 // Cannot be used for the root cgroup.
 Result<Bytes> max(const std::string& cgroup);
 
+
+// Set the soft memory limit for a cgroup and its descendants. Exceeding the
+// soft limit will cause processes in the cgroup to be throttled and put under
+// heavy memory pressure.
+// If limit is None, then there is no soft memory limit.
+// Note: See the top-level `cgroups2::memory` comment about byte alignment.
+//
+// Cannot be used for the root cgroup.
+Try<Nothing> set_soft_max(
+    const std::string& cgroup, const Option<Bytes>& limit);
+
+
+// Get the soft memory limit for a cgroup and its descendants.
+// If the returned limit is None, then there is no soft memory limit.
+//
+// Cannot be used for the root cgroup.
+Result<Bytes> soft_max(const std::string& cgroup);
```

Comment:
let's just call this high and set_high to keep the cgroups terminology intact and avoid any confusion, we don't really want people to have to learn our own terminology here and how it maps to the cgroups terminology

it seems to be called the "memory usage throttle limit" in cgroups v2 and fb's docs FWIW